plugins/services/src/js/components/ServicesQuotaOverviewTable.tsx

```diff
@@ -1,83 +1,63 @@
-import { iconSizeXs } from "@dcos/ui-kit/dist/packages/design-tokens/build/js/designTokens";
+import { Trans } from "@lingui/macro";
+import { Column, Table, SortableHeaderCell } from "@dcos/ui-kit";
 import sort from "array-sort";
 
-import Loader from "#SRC/js/components/Loader";
+import Service from "../structs/Service";
+import ServiceTree from "../structs/ServiceTree";
 
-import { ServiceGroup } from "../types/ServiceGroup";
-import { nameRenderer } from "../columns/QuotaOverviewNameColumn";
-import { limitRenderer } from "../columns/QuotaOverviewLimitColumn";
-import { cpuRenderer } from "../columns/QuotaOverviewCPUConsumedColumn";
-import { memRenderer } from "../columns/QuotaOverviewMemoryConsumedColumn";
-import { diskRenderer } from "../columns/QuotaOverviewDiskConsumedColumn";
-import { gpuRenderer } from "../columns/QuotaOverviewGPUConsumedColumn";
+import { nameRenderer } from "../columns/ServicesTableNameColumn";
+import {
+  limitRenderer,
+  getLimitInfoForService
+} from "../columns/QuotaOverviewLimitColumn";
 
 import { SortDirection } from "../types/SortDirection";
 
 export interface ServicesQuotaOverviewTableProps {
-  groups: ServiceGroup[];
+  serviceTree: ServiceTree;
 }
 
-interface ServicesQuotaOverViewTableState {
-  groups: ServiceGroup[];
+interface ServicesQuotaOverviewTableState {
+  items: Array<Service | ServiceTree>;
   sortDirection: SortDirection;
   sortColumn: string;
 }
 
-const compatatorFor = (prop: string) => (a: ServiceGroup, b: ServiceGroup) =>
-  ServiceGroup.getQuotaPercentage(a, prop) -
-  ServiceGroup.getQuotaPercentage(b, prop);
-
 function sortForColumn(
   columnName: string
-): (a: ServiceGroup, b: ServiceGroup) => number {
+): (a: Service | ServiceTree, b: Service | ServiceTree) => number {
   switch (columnName) {
     case "name":
-      return (a, b) => a.name.localeCompare(b.name);
-    case "limit":
-      return (a, b) =>
-        (a.quota ? a.quota.limitStatus : "N/A").localeCompare(
-          b.quota ? b.quota.limitStatus : "N/A"
-        );
-    case "cpus":
-      return compatatorFor("cpus");
-    case "mem":
-      return compatatorFor("memory");
-    case "disk":
-      return compatatorFor("disk");
-    case "gpus":
-      return compatatorFor("gpus");
+      return (a, b) => a.getName().localeCompare(b.getName());
     default:
-      return () => 0;
+      return (a, b) => {
+        const aText = getLimitInfoForService(a).limitText;
+        const bText = getLimitInfoForService(b).limitText;
+
+        return aText.localeCompare(bText);
+      };
   }
 }
 
 class ServicesQuotaOverviewTable extends React.Component<
   ServicesQuotaOverviewTableProps,
-  ServicesQuotaOverViewTableState
+  ServicesQuotaOverviewTableState
 > {
   constructor(props: Readonly<ServicesQuotaOverviewTableProps>) {
     super(props);
 
     this.state = {
-      groups: [],
+      items: props.serviceTree.getItems(),
```

Comment:
Should we do a default sort on these items here?

Reply:
Great catch!